### PR TITLE
IGNITE-21032 Ensured all entries in AttributeList are Attributes

### PR DESCRIPTION
**Pull request by simon-greatrix**
This pull request resolves issue: https://github.com/apache/ignite/issues/11045

This change ensures that all elements in a created AttributeList are Attribute instances.

https://issues.apache.org/jira/browse/IGNITE-21032

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java [hunk 1]
```diff
@@ -57,16 +58,19 @@ else if ("invoke".equals(actionName))
...
-        AttributeList list = new AttributeList();
-
+        AttributeList attributeList = new AttributeList();
+        List<Attribute> list = attributeList.asList();
         try {
             for (String attribute : attributes) {
                 Object val = getAttribute(attribute);
-
-                list.add(val);
+                if( val instanceof Attribute ) {
```
Comment (sk0x50) [inline modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:66]: Following https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines#CodingGuidelines-BracesandIndentation this code should look like as follows:
```
            for (String attribute : attributes) {
                Object val = getAttribute(attribute);

                if (val instanceof Attribute)
                    list.add((Attribute) val);
                else
                    list.add(new Attribute(attribute, val));
            }
```

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java [hunk 2]
```diff
@@ -57,16 +58,19 @@ else if ("invoke".equals(actionName))
 
     /** {@inheritDoc} */
     @Override public AttributeList getAttributes(String[] attributes) {
-        AttributeList list = new AttributeList();
-
+        AttributeList attributeList = new AttributeList();
```
Comment (sk0x50) [inline modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:61]: Please change `attributeList` to `attrList`
https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules

**Comment by sk0x50** [comment]
I created the following ticket https://issues.apache.org/jira/browse/IGNITE-21032 and moved it to the `In Progress` state. Please assign it to yourself.

**Comment by simon-greatrix** [comment]
@NSAmelchev @sk0x50 - Unfortunately I cannot get access to the Ignite Jira ticket in order to progress it, due to my Apache account being broken. The "password reset" link on the ASF website does not result in me being sent an e-mail, and I can't create an account with that e-mail as it is already in us. I've e-mailed the relevant addresses at ASF but I have not had any response. Until I can progress the Jira, I can't do much.

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java [hunk 3]
```diff
@@ -57,16 +58,18 @@ else if ("invoke".equals(actionName))
 
     /** {@inheritDoc} */
     @Override public AttributeList getAttributes(String[] attributes) {
-        AttributeList list = new AttributeList();
-
+        AttributeList attrList = new AttributeList();
+        List<Attribute> list = attrList.asList();
         try {
-            for (String attr : attributes) {
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:63]: Please let's keep the old variable name

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java [hunk 4]
```diff
@@ -57,16 +58,18 @@ else if ("invoke".equals(actionName))
...
-                Object val = getAttribute(attr);
-
-                list.add(val);
+            for (String attribute : attributes) {
+                Object val = getAttribute(attribute);
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:65]: Let's keep the line break

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java [hunk 5]
```diff
@@ -57,16 +58,18 @@ else if ("invoke".equals(actionName))
...
+                if (val instanceof Attribute)
+                    list.add((Attribute) val);
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:67]: Unnecessary space: `list.add((Attribute)val);`

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java (hunk 4 above)
Comment (simon-greatrix) [reply modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:65]: The line break violates the Ignite coding guide lines. Under "whitespace and empty lines" it says

"Note that single blank line SHOULD NOT be used for the mere decoration of the source code."

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java (hunk 3 above)
Comment (simon-greatrix) [reply modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:63]: Fine. Old name wasn't very good :) - it really should have been "name" to distinguish between the attribute instance, the attribute value and the attribute name.

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java (hunk 5 above)
Comment (simon-greatrix) [reply modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java]: Sapce removed.

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java (hunk 4 above)
Comment (NSAmelchev) [reply modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java]: See table above. The section `Between minimal semantic units in the method or initializer body` says to use 1 line

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java [hunk 6]
```diff
@@ -58,12 +59,14 @@ else if ("invoke".equals(actionName))
     /** {@inheritDoc} */
     @Override public AttributeList getAttributes(String[] attributes) {
         AttributeList list = new AttributeList();
-
+        List<Attribute> attrList = list.asList();
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:62]:
```suggestion
        List<Attribute> attrList = list.asList();

```

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java [hunk 7]
```diff
@@ -58,12 +59,15 @@ else if ("invoke".equals(actionName))
...
     @Override public AttributeList getAttributes(String[] attributes) {
         AttributeList list = new AttributeList();
+        List<Attribute> attrList = list.asList();
 
         try {
             for (String attr : attributes) {
                 Object val = getAttribute(attr);
-
-                list.add(val);
+                if (val instanceof Attribute)
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java:67]:
```suggestion

                if (val instanceof Attribute)
```

File: modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java (hunk 4 above)
Comment (simon-greatrix) [reply modules/core/src/main/java/org/apache/ignite/spi/metric/jmx/ReadOnlyDynamicMBean.java]: Oh wow - that makes for a lot of blank lines! Well if that is the way it is done here :)

**Review by NSAmelchev — APPROVED** [review]

**Comment by NSAmelchev** [comment]
LGTM. I have run [CI](https://tcbot2.sbt-ignite-dev.ru/pr.html?serverId=apache&suiteId=IgniteTests24Java8_RunAll&branchForTc=pull/11046/head&action=Latest).

@sk0x50  Will you take a look too?

**Review by sk0x50 — APPROVED** [review]

**Comment by sk0x50** [comment]
Hello @simon-greatrix @NSAmelchev ,

I merged this patch into the master branch. Thank you for your efforts!

@NSAmelchev could you please cherry-pick the patch into Ignite 2.16?

**Pull request merged**